File: app/src/am_viennv/com/example/asian/UserAdapter.java
package com.example.asian;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.asian.beans.User;

import java.util.List;
import java.util.Locale;

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.ViewHolder> {

    Context mContext;
    List<User> mUserList;
    private final IClickItemUser mIClickItemUser;

    public UserAdapter(IClickItemUser iClickItemUser) {
        this.mIClickItemUser = iClickItemUser;
    }

    public interface IClickItemUser {
        void deleteUser(User user);
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setData(Context context, List<User> userList) {
        this.mContext = context;
        this.mUserList = userList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_user, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        if (mUserList != null && mUserList.size() != 0) {
            User user = mUserList.get(position);
            holder.mTextViewUserId.setText(String.format(Locale.getDefault(), "%d", user.getMIdUser()));
            holder.mTextViewUserName.setText(user.getMUserName());
            holder.mTextViewUserAge.setText(String.format(Locale.getDefault(), "%d", user.getMAge()));

            holder.mButtonDelete.setOnClickListener(view -> mIClickItemUser.deleteUser(user));
        }
    }

    @Override
    public int getItemCount() {
        return mUserList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView mTextViewUserId, mTextViewUserName, mTextViewUserAge;
        private final Button mButtonDelete;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            mTextViewUserId = itemView.findViewById(R.id.tvUserId);
            mTextViewUserName = itemView.findViewById(R.id.tvUserName);
            mTextViewUserAge = itemView.findViewById(R.id.tvUserAge);
            mButtonDelete = itemView.findViewById(R.id.btnDelete);
        }
    }
}

File: Kotlin/room/app/src/main/java/com/example/roomdb/view/UserAdapter.kt
package com.example.roomdb.view

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.recyclerview.widget.RecyclerView
import com.example.roomdb.databinding.ItemUserBinding
import com.example.roomdb.model.User
import com.example.roomdb.viewmodel.UserViewModel

class UserAdapter(private val userViewModel: UserViewModel) : RecyclerView.Adapter<UserAdapter.UserViewHolder>() {

    private var users = MutableLiveData<List<User>>()


    fun setData(list: List<User>) {
        users.postValue(list)
        notifyDataSetChanged()
    }

    class UserViewHolder(private val binding: ItemUserBinding) : RecyclerView.ViewHolder(binding.root) {
        var deleteButton = binding.btnDelete
        fun bind(user: User) {
            binding.tvUserId.text = user.id.toString()
            binding.tvUserName.text = user.name
            binding.tvUserAge.text = user.age.toString()
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        val binding = ItemUserBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return UserViewHolder(binding)
    }

    override fun getItemCount(): Int {
        return users.value?.size ?: 0
    }

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        users.value?.let { holder.bind(it[position]) }
        holder.deleteButton.setOnClickListener {
            users.value?.let { it1 -> deleteUser(it1[position]) }
        }
    }

    @SuppressLint("NotifyDataSetChanged")
    private fun deleteUser(user: User) {
        userViewModel.deleteUser(user)
        notifyDataSetChanged()
    }
}